src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCloneHistory = [];
+  const stateModified = {
+    ...state,
+  };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(stateModified, { ...action.extraData });
+      stateCloneHistory.push({ ...stateModified });
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete stateModified[key];
+      }
+      stateCloneHistory.push({ ...stateModified });
+    }
+
+    if (action.type === 'clear') {
+      for (const key in stateModified) {
+        delete stateModified[key];
+      }
+      stateCloneHistory.push({ ...stateModified });
+    } // else {
+    // stateCloneHistory.push({...stateModified});
+    //
+
+    // }
```

Comment:
Consider uncommenting this section to handle unknown action types. This would ensure that the state history is updated even when an action type is not recognized, maintaining a complete history of state changes.